src/Instrumentation/Symfony/Messenger/InstrumentationEventSubscriber.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace FriendsOfOpenTelemetry\OpenTelemetryBundle\Instrumentation\Symfony\Messenger;
+
+use FriendsOfOpenTelemetry\OpenTelemetryBundle\Instrumentation\InstrumentationTypeEnum;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\API\Trace\TracerInterface;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\SDK\Trace\Span;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageHandledEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageReceivedEvent;
+use Symfony\Component\Messenger\Stamp\BusNameStamp;
+
+/**
+ * Be aware if you start a span before this subscriber, it could leads to orphan span issue.
+ * Be sure your span is properly ended.
+ */
+class InstrumentationEventSubscriber implements EventSubscriberInterface
```

Comment:
Since this subscriber is about worker events, I think we should name it `WorkerMessageEventSubscriber`.